lib/actions/app.coffee

```diff
@@ -44,7 +45,13 @@ exports.create =
 					return callback(new Error('You already have an application with that name!'))
 
 				return callback(null, options.type) if options.type?
-				visuals.patterns.selectDeviceType(callback)
+				resin.models.device.getSupportedDeviceTypes().then (supportedDeviceTypes) ->
+					form.ask
+						message: 'Device Type'
+						name: 'deviceType'
+						type: 'list'
+						choices: supportedDeviceTypes
+					.nodeify(callback)
```

Comment:
It'd be possible, but that means changing two places instead of one. By moving `.nodeify()` to the outer function, you only use it one and keep everything inside a promise chain.